### PR TITLE
MULE-10214: @Import should not require a class element

### PR DESCRIPTION
**Pull request by alegmarra**

### CODE REVIEW

File: modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/capability/xml/schema/ClasspathBasedDslContext.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  }
+
+
+  private void findExtensionsInClasspath() {
+
+    // default classpath for testing environment, were the extension classloader is not useful
+    Set<Class<?>> annotated = getExtensionTypes(forJavaClassPath());
+
+    if (annotated.isEmpty()) {
+      // we are building an standalone extension, thus the classloader of the extension is required
```
Comment (marianogonzalez) [inline modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/capability/xml/schema/ClasspathBasedDslContext.java]: i would prefer to do this search first and do the testing one in a second phase. But let me know if this is not possible.

File: modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/DefaultDslContext.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+import java.util.Optional;
+
+/**
+ * Default implementation of {@link DslResolvingContext} that uses
+ * the {@link ExtensionManager} to provide the required {@link ExtensionModel}s
+ *
+ * @since 4.0
+ */
+class DefaultDslContext implements DslResolvingContext {
```
Comment (marianogonzalez) [inline modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/DefaultDslContext.java]: just to clarify, this is the one that will be used when an application is being deployed on a real mule right?

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/manager/ExtensionEntityKey.java [hunk 3]
```diff
@@ -6,7 +6,7 @@
  */
 package org.mule.runtime.module.extension.internal.manager;
 
-import org.mule.runtime.extension.api.annotation.Extension;
+import static com.sun.imageio.plugins.jpeg.JPEG.vendor;
```
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/manager/ExtensionEntityKey.java]: wtf?

File: modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/DefaultDslContext.java (hunk 2 above)
Comment (alegmarra) [inline modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/DefaultDslContext.java]: Yes, it's the one used by the `ExtensionBuildingDefinitionProvider`.
Other context are: 
- ClasspathBased, used during schema generation at compile time
- SchemaTest, used for testing only

File: modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/capability/xml/schema/ClasspathBasedDslContext.java (hunk 1 above)
Comment (alegmarra) [inline modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/capability/xml/schema/ClasspathBasedDslContext.java]: Having the test classpath first and falling back to this case adds a penalty of >100ms during the compilation of the Http extension, while avoiding a 300 lines warning trace for each test that requires this classpath.

**Pull request merged**